cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/batch/BatchSource.java

```diff
@@ -41,4 +46,28 @@ public void configure(StageConfigurer configurer) {
    * @param context {@link BatchSourceContext}
    */
   public abstract void prepareJob(BatchSourceContext context);
+
+  protected long parseFrequency(String frequencyStr) {
+    //TODO: replace with TimeMathParser (available only internal to cdap)
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(frequencyStr));
+    frequencyStr = frequencyStr.toLowerCase();
+
+    String value = frequencyStr.substring(0, frequencyStr.length() - 1);
+    Preconditions.checkArgument(StringUtils.isNumeric(value));
+    Integer parsedValue = Integer.valueOf(value);
```

Comment:
Should use `Integer.parseInt`. That will through NumberFormatException if it is not a number.